manifests/init.pp

```diff
@@ -56,21 +58,25 @@
     }
   }
 
-  package { 'ipmitool':
-    ensure => $package_ensure,
-    name   => 'ipmitool',
-  }
+  if $manage_package {
+    package { 'ipmitool':
+      ensure => $package_ensure,
+      name   => 'ipmitool',
+    }
 
-  package { 'ipmidriver':
-    ensure => $package_ensure,
-    name   => $openipmi,
+    package { 'ipmidriver':
+      ensure => $package_ensure,
+      name   => $openipmi,
+    }
   }
 
-  service { $service_name:
-    ensure     => $service_ensure,
-    enable     => true,
-    hasrestart => true,
-    hasstatus  => true,
-    require    => [ Package['ipmitool'], Package['ipmidriver']],
+if $manage_service {
+    service { $service_name:
+      ensure     => $service_ensure,
+      enable     => true,
+      hasrestart => true,
+      hasstatus  => true,
+      subscribe  => [ Package['ipmitool'], Package['ipmidriver'], File['ipmiconfig'] ],
```

Comment:
The file should notify the service as this would fail if the file resource did not exist as it does with the default case.  Please switch notify/subscribe back to previous implementation.  Although I do like the file cleanup you did.